SU-HSDO/suhumsci
# SHS-5675: Helping users avoid 'Table cell missing context (e.g. headings) A11y warning

# READY FOR REVIEW

## Summary
- When adding a table to a WYSIWYG field with CKeditor, table headings are now added by default.
- This can be turned off using the table function built into CKEditor, if needed.
- Prevents accessibility tools from flagging poor accessibility due to no table header.

## Need Review By (Date)
7/22/2024

## Urgency
low

## Steps to Test
1. Login to the site and add a flexible page
2. In the Text Area field, add a table using the table tools

<img width="1208" alt="Screenshot 2024-07-16 at 5 02 19 PM" src="https://github.com/user-attachments/assets/01eed7a4-cfb0-4c1a-a76e-79317dcf3539">

3. When the table is created, verify that there is a header row, marked with a dark red background.

<img width="1163" alt="Screenshot 2024-07-16 at 5 02 32 PM" src="https://github.com/user-attachments/assets/177b2ef1-b4eb-4038-9b8a-4ebe3502792b">

4. Enter some data into the table, and save the flexible page.
5. On the front-end, verify that the table header is styled appropriately.

<img width="1287" alt="Screenshot 2024-07-16 at 5 04 39 PM" src="https://github.com/user-attachments/assets/6cc68242-3f7f-42e7-9994-e6b97bb4f675">

@cienvaras - The table is styled correctly when saved on the front-end. However, on the back-end when editing the table, the header text color is too dark against the dark red background so it is hard to see:

![Screenshot 2024-07-16 at 4 21 15 PM](https://github.com/user-attachments/assets/1d2d6d86-8eb4-413a-a9df-6ff668ed1f16)

I know you mentioned to me where the CSS is handled but I've forgotten. I'm happy to fix this if you'd like or if you or Mari could take it on and let me know where the change goes? I would like to know for future tickets. Thank you!

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207835764616004
